### PR TITLE
Migrate rustdoc target spec json path

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/121876.

r? @jieyouxu

### CODE REVIEW

File: tests/run-make/rustdoc-target-spec-json-path/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+use run_make_support::{rustc, rustdoc, tmp_dir};
+
+fn main() {
+    let out_dir = tmp_dir().join("rustdoc-target-spec-json-path");
+    rustc().crate_type("lib").input("dummy_core.rs").target("target.json").run();
+    rustdoc()
+        .input("my_crate.rs")
+        .output(out_dir)
+        .arg("-L")
+        .arg(tmp_dir())
```
Comment (jieyouxu) [inline tests/run-make/rustdoc-target-spec-json-path/rmake.rs:12]: I like

```rs
/// Add a directory to the library search path.
pub fn library_search_path<P: AsRef<Path>>(&mut self, path: P) -> &mut Self {
    self.cmd.arg("-L");
    self.cmd.arg(path.as_ref());
    self
}
```

introduced in https://github.com/rust-lang/rust/pull/125031 for `rustc`. Maybe we should add it for `rustdoc` too?

See https://github.com/rust-lang/rust/pull/125031/files#r1598288585 for discussion.

**Review by jieyouxu — APPROVED** [review]
Thanks! One suggestion for introducing a `library_search_path` helper, then r=me after CI is green.

File: tests/run-make/rustdoc-target-spec-json-path/rmake.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply tests/run-make/rustdoc-target-spec-json-path/rmake.rs:12]: Oh, it's convenient! Adding it to rustdoc then. :)

File: src/tools/run-make-support/src/rustdoc.rs [hunk 2]
```diff
@@ -137,6 +143,13 @@ impl Rustdoc {
         self
     }
 
+    /// Add a directory to the library search path.
```
Comment (jieyouxu) [inline src/tools/run-make-support/src/rustdoc.rs:146]: Can you also note that it corresponds to the rustdoc `-L` option (for a search term)?

Comment (GuillaumeGomez) [reply src/tools/run-make-support/src/rustdoc.rs]: Sure!

**Pull request merged**